validator/schemas/dashboard-schema-v1.json

```diff
@@ -0,0 +1,229 @@
+                              "type": "string",
+                              "title": "The id of the visualization"
+                            }
+                          },
+                          "additionalProperties": true
+                        },
+                        "layout": {
+                          "$id": "#/properties/pages/items/anyOf/0/properties/widgets/items/anyOf/0/properties/layout",
+                          "type": "object",
+                          "title": "The layout of the widget",
+                          "required": [
+                            "column",
+                            "row",
+                            "height",
+                            "width"
+                          ],
+                          "properties": {
+                            "column": {
+                              "$id": "#/properties/pages/items/anyOf/0/properties/widgets/items/anyOf/0/properties/layout/properties/column",
+                              "type": "integer",
+                              "title": "The column position"
+                            },
+                            "row": {
+                              "$id": "#/properties/pages/items/anyOf/0/properties/widgets/items/anyOf/0/properties/layout/properties/row",
+                              "type": "integer",
+                              "title": "The row position"
+                            },
+                            "height": {
+                              "$id": "#/properties/pages/items/anyOf/0/properties/widgets/items/anyOf/0/properties/layout/properties/height",
+                              "type": "integer",
+                              "title": "The height"
+                            },
+                            "width": {
+                              "$id": "#/properties/pages/items/anyOf/0/properties/widgets/items/anyOf/0/properties/layout/properties/width",
+                              "type": "integer",
+                              "title": "The width"
+                            }
+                          },
+                          "additionalProperties": true
+                        },
+                        "title": {
+                          "$id": "#/properties/pages/items/anyOf/0/properties/widgets/items/anyOf/0/properties/title",
+                          "type": "string",
+                          "title": "The title of the widget"
+                        },
+                        "rawConfiguration": {
+                          "$id": "#/properties/pages/items/anyOf/0/properties/widgets/items/anyOf/0/properties/rawConfiguration",
+                          "type": "object",
+                          "title": "The configuration of the widget",
+                          "required": [
+                                "nrqlQueries"
+                              ],
+                          "not": { "required": [ "metrics" ] },
```

Comment:
We've confirmed that it's not possible to filter metric-format widgets by entity guid, which we need for the explorer, so we should not allow dashboards that contain widgets in this format. The query builder now generates NRQL-format metrics so this should only show up if people export old dashboards and we can help them fix it.